config/src/inline/conf_parser.rs

```diff
@@ -0,0 +1,129 @@
+use std::{num::ParseIntError, str::ParseBoolError};
+
+use regex::Regex;
+
+/// Errors returned by the [`ConfParser`] trait.
+#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
+pub enum ConfParserError {
+    /// An invalid configuration property has been provided.
+    /// The property cannot be mapped to the configuration object
+    #[error("'{0}' is not a valid config property")]
+    InvalidConfigProperty(String),
+    /// An error occurred while trying to parse an integer configuration value
+    #[error(transparent)]
+    ParseIntError(#[from] ParseIntError),
+    /// An error occurred while trying to parse a boolean configuration value
+    #[error(transparent)]
+    ParseBoolError(#[from] ParseBoolError),
+}
+
+/// This trait is intended to parse configurations from
+/// structured text. Foundry users can annotate Solidity test functions,
+/// providing special configs just for the execution of a specific test.
+///
+/// An example:
+///
+/// ```solidity
+/// contract MyTest is Test {
+/// /// forge-config: default.fuzz.runs = 100
+/// /// forge-config: ci.fuzz.runs = 500
+/// function test_SimpleFuzzTest(uint256 x) public {...}
+///
+/// /// forge-config: default.fuzz.runs = 500
+/// /// forge-config: ci.fuzz.runs = 10000
+/// function test_ImportantFuzzTest(uint256 x) public {...}
+/// }
+/// ```
+pub trait ConfParser {
+    /// Returns a prefix that is common to all valid configuration lines.
+    /// That helps the parser to extract correct values out of a text.
+    ///
+    /// An example prefix would be `forge-config:default.fuzz.`.
+    fn config_prefix() -> String;
+
+    /// Returns
+    /// - `Some(Self)` in case some configurations are merged into self.
+    /// - `None` in case there are no configurations that can be applied to self.
+    /// - `Err(ConfParserError)` in case of wrong configuration.
+    fn try_merge<S: AsRef<str>>(&self, text: S) -> Result<Option<Self>, ConfParserError>
+    where
+        Self: Sized + 'static;
+
+    /// Given a configuration `text` returns all available pairs (key, value)
+    /// matching the `config_prefix`
+    fn config_variables<S: AsRef<str>>(text: S) -> Vec<(String, String)> {
```

Comment:
this should be covered by the new tests, right?

Reply:
Yes this covered both by unit tests and integration tests